### PR TITLE
470-fe-add-favorites-option-to-resource-card

### PR DESCRIPTION
**Pull request by NataliaR-BCN**

### CODE REVIEW

**Review by JuditKaramazov — APPROVED** [review]
Amazing job!

File: front/src/components/molecules/FavoritesWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from 'react'
```
Comment (kevinmamaqi) [inline front/src/components/molecules/FavoritesWidget.tsx:1]: Renombrar a FavoritesIcon/FavoritesHeart...

File: front/src/components/organisms/MyFavoritesList.tsx [hunk 2]
```diff
@@ -83,37 +72,23 @@ const TextDecorationStyled = styled.span`
...
-
 export const MyFavoritesList: FC = () => {
   const { user } = useAuth()
+  const { slug } = useParams()
   const [isRegisterOpen, setIsRegisterOpen] = useState(false)
   const [isLoginOpen, setIsLoginOpen] = useState(false)
   const [isMobile, setIsMobile] = useState(getWindowIsMobile())
 
+  const { isLoading, isError, data } = useGetFavorites(slug)
+  const favoritesData = { data }.data as TFavorites[] | undefined
```
Comment (kevinmamaqi) [inline front/src/components/organisms/MyFavoritesList.tsx:83]: Eliminar linea

File: front/src/components/organisms/MyFavoritesList.tsx [hunk 3]
```diff
@@ -83,37 +72,23 @@ const TextDecorationStyled = styled.span`
...
-    })
-
 export const MyFavoritesList: FC = () => {
   const { user } = useAuth()
+  const { slug } = useParams()
   const [isRegisterOpen, setIsRegisterOpen] = useState(false)
   const [isLoginOpen, setIsLoginOpen] = useState(false)
   const [isMobile, setIsMobile] = useState(getWindowIsMobile())
 
+  const { isLoading, isError, data } = useGetFavorites(slug)
```
Comment (kevinmamaqi) [inline front/src/components/organisms/MyFavoritesList.tsx:82]:
```suggestion
  const { isLoading, isError, data: favoritesData } = useGetFavorites(slug)
```

File: front/src/components/organisms/MyFavoritesList.tsx [hunk 4]
```diff
@@ -153,42 +128,42 @@ export const MyFavoritesList: FC = () => {
...
-      ) : null}
-
-      {data?.length === 0 ? (
-        <StyledText>No tienes recursos favoritos</StyledText>
-      ) : null}
-
-      {error && user && !isLoading ? <p>Algo ha ido mal...</p> : null}
+      {!isLoading &&
+        !isError &&
+        (favoritesData && favoritesData.length > 0 ? (
```
Comment (kevinmamaqi) [inline front/src/components/organisms/MyFavoritesList.tsx:133]: favoritesData?.length ? ...

**Review by kevinmamaqi — CHANGES_REQUESTED** [review]

File: front/src/components/organisms/MyFavoritesList.tsx (hunk 3 above)
Comment (NataliaR-BCN) [reply front/src/components/organisms/MyFavoritesList.tsx:82]: Also fixed in UserProfile.tsx

File: front/src/components/organisms/MyFavoritesList.tsx (hunk 4 above)
Comment (NataliaR-BCN) [reply front/src/components/organisms/MyFavoritesList.tsx:133]: {data && 
        (data?.length > 0 ?
        ...

**Review by JuditKaramazov — APPROVED** [review]

**Review by kevinmamaqi — APPROVED** [review]

**Pull request merged**